bee-ledger/src/workers/consensus/white_flag.rs

```diff
@@ -0,0 +1,298 @@
+                created_amount = created_amount
+                    .checked_add(output.amount())
+                    .ok_or_else(|| Error::CreatedAmountOverflow(created_amount as u128 + output.amount() as u128))?;
+                balance_diffs.amount_add(*output.address(), output.amount())?;
+                balance_diffs.dust_allowance_add(*output.address(), output.amount())?;
+            }
+            output => return Err(Error::UnsupportedOutputKind(output.kind())),
+        }
+    }
+
+    if created_amount != consumed_amount {
+        return Ok(ConflictReason::InputOutputSumMismatch);
+    }
+
+    for (address, diff) in balance_diffs.iter() {
+        if diff.is_dust_mutating() {
+            let mut balance = storage::fetch_balance_or_default(storage, &address)
+                .await?
+                .apply_diff(diff)?;
+
+            if let Some(diff) = metadata.balance_diffs.get(&address) {
+                balance = balance.apply_diff(diff)?;
+            }
+
+            if balance.dust_outputs() > dust_outputs_max(balance.dust_allowance()) {
+                return Ok(ConflictReason::InvalidDustAllowance);
+            }
+        }
+    }
+
+    for (output_id, created_output) in consumed_outputs {
+        metadata.consumed_outputs.insert(
+            output_id,
+            (created_output, ConsumedOutput::new(*transaction_id, metadata.index)),
+        );
+    }
+
+    for (index, output) in essence.outputs().iter().enumerate() {
+        metadata.created_outputs.insert(
+            OutputId::new(*transaction_id, index as u16)?,
+            CreatedOutput::new(*message_id, output.clone()),
+        );
+    }
+
+    metadata.balance_diffs.merge(balance_diffs)?;
+
+    Ok(ConflictReason::None)
+}
+
+async fn apply_transaction<B: StorageBackend>(
+    storage: &B,
+    message_id: &MessageId,
+    transaction: &TransactionPayload,
+    metadata: &mut WhiteFlagMetadata,
+) -> Result<ConflictReason, Error> {
+    match transaction.essence() {
+        Essence::Regular(essence) => {
+            apply_regular_essence(
+                storage,
+                message_id,
+                &transaction.id(),
+                essence,
+                transaction.unlock_blocks(),
+                metadata,
+            )
+            .await
+        }
+        essence => Err(Error::UnsupportedTransactionEssenceKind(essence.kind())),
+    }
+}
+
+async fn apply_message<B: StorageBackend>(
+    storage: &B,
+    message_id: &MessageId,
+    message: &Message,
+    metadata: &mut WhiteFlagMetadata,
+) -> Result<(), Error> {
+    metadata.referenced_messages += 1;
+
+    match message.payload() {
+        Some(Payload::Transaction(transaction)) => {
+            match apply_transaction(storage, message_id, transaction, metadata).await? {
+                ConflictReason::None => metadata.included_messages.push(*message_id),
+                conflict => metadata.excluded_conflicting_messages.push((*message_id, conflict)),
```

Comment:
Same here

Reply:
How would I get the conflict on the else branch with an if let ?